Support serialization to NativeFontHandle

alfredoyang opened the pull request:
NativeFontHandle needs to implement serialization due to it is removed from ```CGFont``` in ```core-graphics```.

https://github.com/servo/servo/issues/15607

This PR depends on https://github.com/servo/core-graphics-rs/pull/82.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/webrender/1133)
<!-- Reviewable:end -->

alfredoyang commented:
The failure is due to it requires https://github.com/servo/core-graphics-rs/pull/82.

kvark commented on webrender_traits/Cargo.toml:
```diff
@@ -13,6 +13,7 @@ webgl = ["offscreen_gl_context"]
 [dependencies]
 app_units = "0.4"
 byteorder = "1.0"
+core-foundation = "0.3"
```
we could make it an OSX only dependency (via `
[target.'cfg(osx)'.dependencies]`)

kvark commented on webrender_traits/src/font.rs:
```diff
@@ -6,12 +6,36 @@ use app_units::Au;
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error> where D: Deserializer {
+        let postscript_name: String = try!(Deserialize::deserialize(deserializer));
+        CGFont::from_name(&CFString::new(&*postscript_name))
+            .map(|font| NativeFontHandle(font))
```
this could look cleaner with a simple `match` instead of `map`+`map_err` pair

alfredoyang commented:
hmm...
It looks like I forgot to add ```#[cfg(target_os = "macos")]``` before
```impl Deserialize for NativeFontHandle```.

alfredoyang commented:
Review status: 0 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[webrender_traits/Cargo.toml, line 16 at r1](https://reviewable.io:443/reviews/servo/webrender/1133#-Ki-XTUj9bsv6-Y4iCQ8-r1-16:-Ki4Wjh6R0WIgsCXnxoY:bt2vr1b) ([raw file](https://github.com/servo/webrender/blob/c52148f41e783614fec4693c3c53abf02f91ce8a/webrender_traits/Cargo.toml#L16)):*
<details><summary><i>Previously, kvark (Dzmitry Malyshau) wrote…</i></summary><blockquote>

we could make it an OSX only dependency (via `
[target.'cfg(osx)'.dependencies]`)
</blockquote></details>

Fixed.

---

*[webrender_traits/src/font.rs, line 32 at r1](https://reviewable.io:443/reviews/servo/webrender/1133#-Ki-XTUj9bsv6-Y4iCQ9-r1-32:-Ki4WoysCSlXaC7k1SSx:bt2vr1b) ([raw file](https://github.com/servo/webrender/blob/c52148f41e783614fec4693c3c53abf02f91ce8a/webrender_traits/src/font.rs#L32)):*
<details><summary><i>Previously, kvark (Dzmitry Malyshau) wrote…</i></summary><blockquote>

this could look cleaner with a simple `match` instead of `map`+`map_err` pair
</blockquote></details>

Fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/webrender/1133)*
<!-- Sent from Reviewable.io -->

kvark commented:
Thanks @alfredoyang ! Do you mind squashing the commits before we merge?

alfredoyang commented:
Please squash them. Btw, I can ```git --amend``` next time if it is the preferred way.

KiChjang commented:
@alfredoyang I think @kvark meant whether you mind squashing your own commits, before the PR is merged.

alfredoyang commented:
Ok, it's done.

kvark commented:
Thank you!
@bors-servo r+

bors-servo commented:
:pushpin: Commit e83ff76 has been approved by `kvark`

<!-- @bors-servo r=kvark e83ff767a93bc6394d2f8dccabdd8b3df64e9a1b -->

bors-servo commented:
:hourglass: Testing commit e83ff767a93bc6394d2f8dccabdd8b3df64e9a1b with merge f3fa3481aac63ac93c6ccbe805379875e24e5b77...

bors-servo commented:
:sunny: Test successful - [status-travis](https://travis-ci.org/servo/webrender/builds/223807302)
Approved by: kvark
Pushing f3fa3481aac63ac93c6ccbe805379875e24e5b77 to master...
